SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/EditorWorkerController.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma once
+
+#if WITH_EDITOR
+
+#include "Editor.h"
+#include "Settings/LevelEditorPlaySettings.h"
+
+namespace improbable
+{
+
+struct EditorWorkerController
+{
+	void OnPrePIEEnded(bool bValue)
+	{
+		LastPIEEndTime = FDateTime::Now().ToUnixTimestamp();
+		FEditorDelegates::PrePIEEnded.Remove(PIEEndHandle);
+	}
+
+	void InitWorkers(const FString& WorkerType)
+	{
+		ReplaceProcesses.Empty();
+
+		// Only issue the worker replace request if there's a chance the load balancer hasn't acknowledged
+		// that the previous session's workers have disconnected. There's no hard `heartbeat` time for this as
+		// it's dependent on multiple factors (fabric load etc.), so this value was landed on after significant
+		// trial and error.
+		const int64 WorkerReplaceThresholdSeconds = 8;
+
+		int64 SecondsSinceLastSession = FDateTime::Now().ToUnixTimestamp() - LastPIEEndTime;
+		UE_LOG(LogSpatialWorkerConnection, Verbose, TEXT("Seconds since last session - %d"), SecondsSinceLastSession);
+
+		PIEEndHandle = FEditorDelegates::PrePIEEnded.AddRaw(this, &EditorWorkerController::OnPrePIEEnded);
+
+		int32 PlayNumberOfServers;
+		GetDefault<ULevelEditorPlaySettings>()->GetPlayNumberOfServers(PlayNumberOfServers);
+
+		WorkerIds.SetNum(PlayNumberOfServers);
+		for (int i = 0; i < PlayNumberOfServers; ++i)
+		{
+			FString NewWorkerId = WorkerType + FGuid::NewGuid().ToString();
+
+			if (!WorkerIds[i].IsEmpty() && SecondsSinceLastSession < WorkerReplaceThresholdSeconds)
+			{
+				ReplaceProcesses.Add(ReplaceWorker(WorkerIds[i], NewWorkerId));
+			}
+
+			WorkerIds[i] = NewWorkerId;
+		}
+
+	}
+
+	FProcHandle ReplaceWorker(const FString& OldWorker, const FString& NewWorker)
+	{
+		const FString CmdExecutable = TEXT("spatial.exe");
+
+		const FString CmdArgs = FString::Printf(
+			TEXT(" local worker replace "
+				"--existing_worker_id %s "
+				"--replacing_worker_id %s"), *OldWorker, *NewWorker);
+		uint32 ProcessID = 0;
+		FProcHandle ProcHandle = FPlatformProcess::CreateProc(
```

Comment:
Is it okay to call multiple `spatial local worker replace` at the same time?

Reply:
Locally I've tested calling them in parallel, and it seems fine. Have asked for confirmation though.